.travis/sitecustomize.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python2
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# This file is used by Travis-CI to start the coverage process. In order to make
+# Cylc and Python aware of it, we export PYTHONPATH when running the tests
+
+import coverage
+coverage.process_startup()
```

Comment:
What calls this `sitecustomize.py` script?

Reply:
We are setting the `PYTHONPATH` variable in the `.travis.yml` script. When Python finds that script in the `PYTHONPATH`, it loads it (similarly to those `.pth` files). Here's the recent docs, but that applies to Python 2.x too : https://docs.python.org/3/library/site.html (search for sitecustomize).

Handy trick, that we had to use because the coverage module needs to be aware that we want it to run even though what started it was a shell script in the functional tests with `test-battery`.

Reply:
Cool, diddn't know about that one!